File: Recursion/Fibo/fibo.py
def fibo(number):
    cache = {}
    def fibonacci(number):
        if number < 2:
            return 1
        if number in cache:
            return cache[number]
        return fibo(number-1)+fibo(number-2)
    
    return  fibonacci(number)
   
    
example = fibo(15) # expected 89
print(example)

File: Recursion/Fibo/fibo.ts
function fibon(number:number):number {
    if (number < 2) {
      return 1;
    }
    return fibon(number - 1) + fibon(number - 2);
  }
  
  let examp = fibon(10); // expected 89
  console.log(examp);
  